framework/console/Request.php

```diff
@@ -66,10 +66,15 @@ public function resolve()
 
         $params = [];
         foreach ($rawParams as $param) {
-            if (preg_match('/^--(\w+)(=(.*))?$/', $param, $matches)) {
+            if (preg_match('/^--(\w+)(?:=(.*))?$/', $param, $matches)) {
                 $name = $matches[1];
                 if ($name !== Application::OPTION_APPCONFIG) {
-                    $params[$name] = isset($matches[3]) ? $matches[3] : true;
+                    $params[$name] = isset($matches[2]) ? $matches[2] : true;
+                }
+            } elseif (preg_match('/^-(\w+)(?:=(.*))?$/', $param, $matches)) {
+                $name = $matches[1];
+                if ($name !== Application::OPTION_APPCONFIG) {
```

Comment:
```-config``` works only if you add to optionsAliases, if not it show the following message : 

![screenshot_2](https://cloud.githubusercontent.com/assets/6630197/13331988/2d8324d0-dc00-11e5-9039-e8326349f2d8.png)

see https://github.com/yiisoft/yii2/pull/10870/files#diff-3f2a12dd38a81f2b0c96a4c9e6dd7eabR96